jmclient/jmclient/blockchaininterface.py

```diff
@@ -285,18 +285,34 @@ def import_addresses_if_needed(self, addresses, wallet_name):
             self.import_addresses(addresses - imported_addresses, wallet_name)
         return import_needed
 
-    def _yield_transactions(self, wallet_name):
-        batch_size = 1000
-        iteration = 0
+    def _yield_transactions(self):
+        """ Generates a lazily fetched sequence of transactions seen in the
+        wallet (under any label/account), yielded in newest-first order. Care
+        is taken to avoid yielding duplicates even when new transactions are
+        actively being added to the wallet while the iteration is ongoing.
+        """
+        num, skip = 1, 0
+        txs = self.list_transactions(num, skip)
+        if not txs:
+            return
+        tx = txs[0]
+        yield tx
         while True:
-            new = self._rpc(
-                'listtransactions',
-                ["*", batch_size, iteration * batch_size, True])
-            for tx in new:
+            num *= 2
+            txs = self.list_transactions(num, skip)
+            if not txs:
+                return
+            try:
+                idx = [(tx['txid'], tx['vout'], tx['category']) for tx in txs
+                        ].index((tx['txid'], tx['vout'], tx['category']))
+            except ValueError:
+                skip += num
+                continue
+            for tx in reversed(txs[:idx]):
```

Comment:
Probably just dumb but: why `reversed`? I thought `list_transactions` is returning the txs in newest-first order, which is what you want here? So like if the last transaction appears at index 10, txs[:10] would be the ones after it, newest first?

Reply:
`list_transactions` just passes the result of the `listtransactions` RPC straight through. Bitcoin Core's `listtransactions` RPC returns results in oldest-first order. (It's surprising but true.)

Reply:
For example, if you call `listtransactions` with skip=1 and num=2, then you will get a list containing the third-newest transaction at index 0 and the second-newest transaction at index 1.

Reply:
> For example, if you call listtransactions with skip=1 and num=2, then you will get a list containing the third-newest transaction at index 0 and the second-newest transaction at index 1.
Right. It's extremely counterintuitive, so thanks for the reminder of the logic. It returns "last num", but only after skipping "last skip" and returns them in *forward* time order. Sheesh.

So:  listtransactions(num=8,skip=5) means the 13th most recent to the 6th most recent, in that order.

Reply:
I had picked up an older PR for Bitcoin Core that would allow to change this, get it from RPC in an order you want and also skipping by pointer, but it is stalled, mainly due to disagreements how to add new arguments to `listtransactions`. https://github.com/bitcoin/bitcoin/pull/19443